docs/config/index.md

```diff
@@ -123,11 +123,11 @@ export default defineConfig(async ({ command, mode }) => {
 
   Define global constant replacements. Entries will be defined as globals during dev and statically replaced during build.
 
-  - Starting from `2.0.0-beta.70`, string values will be used as raw expressions, so if defining a string constant, it needs to be explicitly quoted (e.g. with `JSON.stringify`).
-
-  - Replacements are performed only when the match is surrounded by word boundaries (`\b`).
+  :::warning String constants
+  Starting from `2.0.0-beta.70`, string values will be used as raw expressions, so if defining a string constant, it needs to be explicitly quoted (e.g. with `JSON.stringify`).
+  :::
 
-  Because it's implemented as straightforward text replacements without any syntax analysis, we recommend using `define` for CONSTANTS only.
+  Because it's implemented as straightforward [RegExp-based](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/define.ts#L62-L76) text replacements without any syntax analysis, we recommend using `define` for CONSTANTS only.
```

Comment:
It's "dangerous" to reference the main branch without permalink, due to the code could change and then the link points to something wrongly

Reply:
Very true, nice catch! Updated :)